pvlib/tests/iotools/test_pvgis.py

```diff
@@ -4,12 +4,245 @@
+            'variables': {
+                'technology': {'description': 'PV technology'},
+                'peak_power': {'description': 'Nominal (peak) power of the PV module', 'units': 'kW'},  # noqa: E501
+                'system_loss': {'description': 'Sum of system losses', 'units': '%'}}}},  # noqa: E501
+    'outputs': {
+        'hourly': {
+            'type': 'time series', 'timestamp': 'hourly averages',
+            'variables': {
+                'P': {'description': 'PV system power', 'units': 'W'},
+                'Gb(i)': {'description': 'Beam (direct) irradiance on the inclined plane (plane of the array)', 'units': 'W/m2'},  # noqa: E501
+                'Gd(i)': {'description': 'Diffuse irradiance on the inclined plane (plane of the array)', 'units': 'W/m2'},  # noqa: E501
+                'Gr(i)': {'description': 'Reflected irradiance on the inclined plane (plane of the array)', 'units': 'W/m2'},  # noqa: E501
+                'H_sun': {'description': 'Sun height', 'units': 'degree'},
+                'T2m': {'description': '2-m air temperature', 'units': 'degree Celsius'},  # noqa: E501
+                'WS10m': {'description': '10-m total wind speed', 'units': 'm/s'},  # noqa: E501
+                'Int': {'description': '1 means solar radiation values are reconstructed'}}}}}  # noqa: E501
+
+
+# Test read_pvgis_hourly function using two different files with different
+# input arguments (to test variable mapping and pvgis_format)
+@pytest.mark.parametrize('testfile,index,columns,values,metadata_exp,'
+                         'inputs_exp,map_variables,pvgis_format', [
+                             (testfile_radiation_csv, index_radiation_csv,
+                              columns_radiation_csv, data_radiation_csv,
+                              metadata_radiation_csv, inputs_radiation_csv,
+                              False, None),
+                             (testfile_radiation_csv, index_radiation_csv,
+                              columns_radiation_csv_mapped, data_radiation_csv,
+                              metadata_radiation_csv, inputs_radiation_csv,
+                              True, 'csv'),
+                             (testfile_pv_json, index_pv_json, columns_pv_json,
+                              data_pv_json, metadata_pv_json, inputs_pv_json,
+                              False, None),
+                             (testfile_pv_json, index_pv_json,
+                              columns_pv_json_mapped, data_pv_json,
+                              metadata_pv_json, inputs_pv_json, True, 'json')])
+def test_read_pvgis_hourly(testfile, index, columns, values, metadata_exp,
+                           inputs_exp, map_variables, pvgis_format):
+    # Create expected dataframe
+    expected = pd.DataFrame(index=index, data=values, columns=columns)
+    expected['Int'] = expected['Int'].astype(int)
+    expected.index.name = 'time'
+    expected.index.freq = None
```

Comment:
Would it be cleaner to define fixtures for the expected dataframes?  pytest can't parametrize directly across fixtures (unfortunately), but it has a `getfixturevalue` function that can be used as a workaround, for example: https://stackoverflow.com/a/64348247/1641381

Reply:
The solution has been implemented and did indeed make the tests more readable and compact.

Reply:
The solution has been implemented and did indeed make the tests more readable and compact.